testcases/OpTestDlparIO.py

```diff
@@ -0,0 +1,289 @@
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2021
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+#
+
+'''
+OpTestDlpar
+---------
+
+This test is to preform and validate dlpar commands on SRIO and Pysical adapters
+'''
+
+import unittest
+import time
+import re
+import OpTestConfiguration
+import OpTestLogger
+from common import OpTestHMC
+from common.OpTestSystem import OpSystemState
+from common.OpTestError import OpTestError
+from common.Exceptions import CommandFailed
+
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+class OpTestDlpar(unittest.TestCase):
+
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.console = self.cv_SYSTEM.console
+        self.cv_HMC = self.cv_SYSTEM.hmc
+        self.cv_HOST = conf.host()
+        self.mg_system = self.cv_HMC.mg_system
+        self.dest_lpar = conf.args.target_lpar
+        self.pci_device = conf.args.pci_device
+        self.sriov = conf.args.sriov
+        self.num_of_dlpar = int(conf.args.num_of_dlpar)
+
+        if not self.cv_HMC.is_lpar_in_managed_system(self.mg_system, self.cv_HMC.lpar_name):
+            raise OpTestError("Lpar %s not found in managed system %s" % (
+                              self.cv_HMC.lpar_name, self.mg_system))
+        if not self.cv_HMC.is_lpar_in_managed_system(self.mg_system, self.dest_lpar):
+            raise OpTestError("Lpar %s not found in managed system %s" % (self.dest_lpar, self.mg_system))
+        self.check_pkg_installation()
+        self.rsct_service_start()
+        if self.dest_lpar is not None:
+            cmd = 'lshwres -r io -m %s --rsubtype slot --filter \
+                   lpar_names=%s -F lpar_id' % (self.mg_system, self.dest_lpar)
+            output = self.cv_HMC.ssh.run_command(cmd)
+            self.dest_lpar_id =  output[0]
+        self.loc_code = self.get_slot_from_sysfs(self.pci_device)
+        self.get_slot_hw_details()
+
+    def check_pkg_installation(self):
```

Comment:
function doc string can be added here to explain the function objective

Reply:
Added